agent/php_execute.c

```diff
@@ -2290,19 +2017,34 @@ static void nr_php_instrument_func_end(NR_EXECUTE_PROTO) {
    * Check if we have special instrumentation for this function or if the user
    * has specifically requested it.
    */
-  wraprec = segment->wraprec;
 
   if (NULL != wraprec) {
     /*
      * This is the case for specifically requested custom instrumentation.
      */
     create_metric = wraprec->create_metric;
 
-    zcaught = nr_zend_call_orig_execute_special(wraprec, segment,
+    /*
+     * A NULL return value means that there was an uncaught exception, and
+     * therefore we want to call the 'clean' function type
+     */
+    if (NULL != nr_php_get_return_value(NR_EXECUTE_ORIG_ARGS TSRMLS_CC)) {
```

Comment:
```suggestion
    if (NULL != nr_php_get_return_value(NR_EXECUTE_ORIG_ARGS)) {
```
`TSRM` macros aren't necessary anymore